contracts/programs/bubblegum/src/lib.rs

```diff
@@ -46,17 +49,79 @@ pub struct InitNonce<'info> {
 
 #[derive(Accounts)]
 pub struct CreateTree<'info> {
+    #[account(mut)]
     pub tree_creator: Signer<'info>,
     #[account(
+        init,
         seeds = [merkle_slab.key().as_ref()],
         bump,
+        space=GUMMYROLL_TREE_AUTHORITY_SIZE + 64,
+        payer=tree_creator,
     )]
     /// CHECK: This account is neither written to nor read from.
-    pub authority: UncheckedAccount<'info>,
+    pub authority: Account<'info, GummyrollTreeAuthority>,
     pub gummyroll_program: Program<'info, Gummyroll>,
     #[account(zero)]
     /// CHECK: This account must be all zeros
     pub merkle_slab: UncheckedAccount<'info>,
+    pub system_program: Program<'info, System>,
+}
+
+#[derive(Accounts)]
+pub struct TransferTreeOwner<'info> {
+    pub tree_owner: Signer<'info>,
+    /// CHECK: this account is not read from
+    pub new_owner: AccountInfo<'info>,
+    #[account(
+        seeds=[authority.tree_id.as_ref()],
+        bump,
+        mut, 
+        constraint = *tree_owner.key == authority.owner
+    )]
+    pub authority: Account<'info, GummyrollTreeAuthority>,
+}
+
+#[derive(Accounts)]
+pub struct TransferTreeDelegate<'info> {
+    pub tree_delegate: Signer<'info>,
+    /// CHECK: this account is not read from
+    pub new_delegate: AccountInfo<'info>,
+    #[account(
+        seeds=[authority.tree_id.as_ref()],
+        bump,
+        mut, 
```

Comment:
Nit: prefer mut as the first constraint, but not a big deal

Reply:
Fixed